### PR TITLE
refactor(console): improve hasUnsavedChanges by leveraging form isDiry state

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

Improve `hasUnsavedChanges` logic in `ConnectorContent` component. Use form `isDirty` state instead.

## Testing

- [x] Tested locally in connector details page. Form submit and unsave navigation block both worked properly

### CODE REVIEW

File: packages/console/src/pages/ConnectorDetails/index.tsx [hunk 1]
```diff
@@ -172,7 +170,6 @@ const ConnectorDetails = () => {
             </TabNavItem>
           </TabNav>
           <ConnectorContent
-            isDeleted={isDeleted}
```
Comment (xiaoyijun) [inline packages/console/src/pages/ConnectorDetails/index.tsx:175]: We should not remove this `isDeleted` logic; this causes a bug when deleting the connector, which has unsaved changes.

Comment (demonzoo) [reply packages/console/src/pages/ConnectorDetails/index.tsx:175]: Wasn't able to reproduce it last night. Maybe I forgot the steps.

Comment (demonzoo) [reply packages/console/src/pages/ConnectorDetails/index.tsx]: You are right. Code updated. @xiaoyijun

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Pull request merged**